src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -789,19 +789,15 @@ public override bool Equals(object? o)
             else g = (Guid)o;
 
             // Now compare each of the elements
-            return g._a == _a &&
-                Unsafe.Add(ref g._a, 1) == Unsafe.Add(ref _a, 1) &&
-                Unsafe.Add(ref g._a, 2) == Unsafe.Add(ref _a, 2) &&
-                Unsafe.Add(ref g._a, 3) == Unsafe.Add(ref _a, 3);
+            return Unsafe.As<int, long>(ref g._a) == Unsafe.As<int, long>(ref _a) &&
+                Unsafe.As<byte, long>(ref g._d) == Unsafe.As<byte, long>(ref _d);
         }
 
         public bool Equals(Guid g)
         {
             // Now compare each of the elements
-            return g._a == _a &&
-                Unsafe.Add(ref g._a, 1) == Unsafe.Add(ref _a, 1) &&
-                Unsafe.Add(ref g._a, 2) == Unsafe.Add(ref _a, 2) &&
-                Unsafe.Add(ref g._a, 3) == Unsafe.Add(ref _a, 3);
+            return Unsafe.As<int, long>(ref g._a) == Unsafe.As<int, long>(ref _a) &&
```

Comment:
Does this logic need to be repeated in multiple places rather than have everything forward to `==` for example?

I'd hope the JIT would properly inline the check in all the cases and the code would be equivalent.

Reply:
Agreed, although that's a separate issue.